14/src/main.c

```diff
@@ -0,0 +1,276 @@
+        ptr = line;
+        while (sscanf(ptr, "%d,%d", &x, &y) == 2) {
+            if (x < min_x) min_x = x;
+            if (y < min_y) min_y = y;
+            if (x > max_x) max_x = x;
+            if (y > max_y) max_y = y;
+
+            ptr = strchr(ptr, '>');
+            if (ptr) ptr += 2;
+            else break;
+        }
+    }
+    fseek(file, 0, SEEK_SET);
+
+    // the x coordinate in the task does not start from 0, but from a number around 500. 
+    // Therefore, we need to set the origin of coordinates for our game to the minimum from "min in task-1", 
+    // in order to leave empty passages for sand on the left and right of the field
+    gameMap->min_x = min_x-1; 
+    gameMap->min_y = 0;
+
+    // when setting the size, the field should be 1 block more to the left and right.
+    // so that the sand wakes up. And the height is greater to consider the bottom line
+    // as empty and complete the task
+    gameMap->width = (max_x - min_x) + 3;
+    gameMap->height = max_y + 3;
+
+    gameMap->map = (char**) malloc(gameMap->height * sizeof(char*));
+    for (int i = 0; i < gameMap->height; i++) {
+        gameMap->map[i] = (char*) malloc(gameMap->width * sizeof(char));
+        for (int j = 0; j < gameMap->width; j++) {
+            gameMap->map[i][j] = '.';
+        }
+    }
+}
+
+
+
+void parse_input(const char *filename, GameMap* gameMap) {
+    FILE* file = fopen(filename, "r");
+    if (!file) {
+        printf("\033[1;31m");
+        perror("Failed to open data.txt");
+        printf("\033[0m"); 
+        exit(1);
+    }
+
+    find_bounds(file, gameMap);
+
+
+    char line[1000];
+    while (fgets(line, sizeof(line), file)) {
+        PointArray* arr = create_point_array(10);
+        int x, y;
+        char* ptr = line;
+        
+        while (sscanf(ptr, "%d,%d", &x, &y) == 2) {
+            add_point(arr, x - gameMap->min_x, y - gameMap->min_y);
+            ptr = strchr(ptr, '>'); 
+            if (ptr) ptr += 2;
+            else break;
+        }
+        
+        for (int i = 0; i < arr->size - 1; i++) {
+            draw_line(gameMap, arr->data[i], arr->data[i + 1]);
+        }
+
+        free(arr->data);
+        free(arr);
+    }
+    
+    fclose(file);
+}
+
+
+int is_cell_empty(GameMap* gameMap, Point p) {
+    if (p.y >= gameMap->height ||
+        p.x < 0 || 
+        p.x >= gameMap->width ||
+        gameMap->map[p.y][p.x] != '.') {
+            return 0;
+    }
+    return 1;
+
+}
+
+Point calculate_next_move(GameMap* gameMap, Point current) {
+    if(current.y == gameMap->height-1) return current;
+
+    Point down = {current.x, current.y + 1};
+    Point down_left = {current.x - 1, current.y + 1};
+    Point down_right = {current.x + 1, current.y + 1};
+
+    if (is_cell_empty(gameMap, down)) return down;
+
+    else if (is_cell_empty(gameMap, down_left)) return down_left;
+    else if (is_cell_empty(gameMap, down_right)) return down_right;
+
+    else return current;
```

Comment:
don't use **else** after return
```suggestion
    if (is_cell_empty(gameMap, down)) return down;
    if (is_cell_empty(gameMap, down_left)) return down_left;
    if (is_cell_empty(gameMap, down_right)) return down_right;

    return current;
```